### PR TITLE
Submeters for METER and LOADMETER

### PR DESCRIPTION
**Pull request by chipbarnaby**
## Description

Initial implementation of submeters for METER and LOADMETER
- addition of METER mtrSubmeters and mtrSubmeterMults
- addition of LOADMETER lmtSubmeters and lmtSubmeterMults
- checking and sorting code that treats cyclic refs as errors and warns about duplicate refs
- associated test file that includes deliberate duplicate references to test dup ref detection and warning
- associated documentation

Also includes incomplete DHWSYSRES accounting fixes.

### CODE REVIEW

File: src/cnguts.cpp [hunk 1]
```diff
@@ -2072,7 +2073,199 @@ LOCAL void FC accumAhr( 		// Accumulate air handler simulation results
...
+
+	// determine submeter accumulation sequences
+	// can fail due to cyclic refs
+	if (rc == RCOK)
+		rc = SubMeterSeq.smsq_Setup();
+
+	return rc;
+}	// cgSubMeterSetup
+//------------------------------------------------------------------------------
+RC SUBMETERSEQ::smsq_Setup()	// derive submeter sequences
```
Comment (nealkruis) [inline src/cnguts.cpp:2240]: smsq? 😉

File: src/dhwcalc.cpp [hunk 2]
```diff
@@ -4663,8 +4688,8 @@ RC DHWHEATER::wh_DoSubhrEnd(		// end-of-subhour
 	{
 		wh_HPWH.hw_DoSubhrEnd(mult, wh_pZn, wh_pAshpSrcZn);
 
-		wh_qLoss = wh_HPWH.hw_qLoss;
-		wh_qEnv = wh_HPWH.hw_qEnv;
+		wh_qLoss = BtuperkWh * wh_HPWH.hw_qLoss;	// tank loss, + = to surround
+		wh_qEnv = BtuperkWh * wh_HPWH.hw_qEnv;
```
Comment (nealkruis) [inline src/dhwcalc.cpp:4692]: Doesn't seem relevant to this PR.

File: src/nummeth.cpp [hunk 3]
```diff
@@ -326,6 +326,94 @@ int regula(							// find x given f(x) (regula-falsi method)
...
+
+	// Depth-first search starting from each vertex
+	for (int iV = 0; iV < dg_nV; iV++)
+	{	if (!dg_TopologicalSortDFS(iV, vSorted))
+			return false;	// cyclic detected, abandon
+	}
+	return true;
+}	// DGRAPH::dg_TopologicalSort
+//-----------------------------------------------------------------------------
+bool DGRAPH::dg_TopologicalSortDFS(
```
Comment (nealkruis) [inline src/nummeth.cpp:350]: DFS?

**Review by nealkruis — APPROVED** [review]
@chipbarnaby just a couple comments on confusing names and unrelated changes.

File: src/cnguts.cpp (hunk 1 above)
Comment (nealkruis) [reply src/cnguts.cpp:2240]: ...nm. This is just a prefix. The context should be (mostly) clear based on the class name.

Comment (chipbarnaby) [reply src/cnguts.cpp:2240]: smsq = submeter sequence

File: src/dhwcalc.cpp (hunk 2 above)
Comment (chipbarnaby) [reply src/dhwcalc.cpp:4692]: It is not.  Work on DHW energy balance became entangled.

File: src/nummeth.cpp (hunk 3 above)
Comment (chipbarnaby) [reply src/nummeth.cpp:350]: DFS = depth first search
